baseplate/experiments/providers/__init__.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import unicode_literals
+
+import logging
+import time
+
+from .forced_variant import ForcedVariantExperiment
+from .legacy import LegacyExperiment
+from ...features.feature import feature_flag_from_config
+
+logger = logging.getLogger(__name__)
+
+
+class ExperimentManager(object):
+
+    def __init__(self, id, name, owner, experiment, enabled=True,
+                 feature_flag=None):
+        self.id = id,
+        self.name = name
+        self.owner = owner
+        self._experiment = experiment
+        self._enabled = enabled
+        self._feature_flag = feature_flag
+
+    def enabled(self, **args):
+        if not self._enabled:
+            return False
+
+        if self._feature_flag is None:
+            return True
+
+        if self._feature_flag.enabled(**args):
+            return True
+
+        return False
+
+    def variant(self, **args):
+        return self._experiment.variant(**args)
+
+    def should_log_bucketing(self):
+        return self._experiment.should_log_bucketing()
+
+    def event_params(self):
+        event_params = {
+            "experiment_id": self.id,
+            "experiment_name": self.name,
+            "experiment_owner": self.owner,
+        }
+        return event_params
+
+
+def experiment_from_config(config):
+    experiment_type = config["type"].lower()
+    experiment_id = config["id"]
+    name = config["name"]
+    owner = config.get("owner")
+    experiment_config = config["experiment"]
+    expiration = config["expires"]
+    if int(time.time()) > expiration:
+        logger.warning(
+            "Found an expired experiment <%s:%s> that is owned by <%s>. "
+            "Please clean up.",
+            experiment_id,
+            name,
+            owner,
+        )
+        return ExperimentManager(
+            id=experiment_id,
+            name=name,
+            owner=owner,
+            experiment=ForcedVariantExperiment(None),
+        )
+    override = config.get("global_override")
+    enabled = config.get("enabled", True)
+    if override:
+        logger.warning(
+            "Found an experiment with a global override <%s:%s> that is owned "
+            "by <%s>. Please clean up.",
+            experiment_id,
+            name,
+            owner,
+        )
+        return ExperimentManager(
+            id=experiment_id,
+            name=name,
+            owner=owner,
+            experiment=ForcedVariantExperiment(override)
+        )
+    if not enabled:
+        logger.warning(
+            "Found a disabled experiment <%s:%s> that is owned by <%s>. "
+            "Please clean up.",
+            experiment_id,
+            name,
+            owner,
+        )
+        return ExperimentManager(
+            id=experiment_id,
+            name=name,
+            owner=owner,
+            experiment=ForcedVariantExperiment(None),
+        )
+
+    if "feature" in config:
+        feature_flag = feature_flag_from_config(config["feature"])
```

Comment:
I've gone back and forth on this. The problem I've ran into is that features just return True if any of their targeting parameters are met while experiments use targeting to determine if they should run an experiment. 

I've thought about it some more I have an idea about how to use experiments to get this behavior. 

1. Make `targeting` generic. The keys in the `targeting` dict are potential `variant` kwarg values. The values are the "targeted" values. They can be either a single value or a list of values.
2. Add a notion of `overrides`. This would be a dict mapping string keys to dict. The keys would be potention kwargs. The inner dicts would map kwarg values to variants. 

A "feature_flag" experiment has one variant "active" and uses `overrides` rather than `targeting` for most of the values.